lib/rules/function-name-arguments-whitelist/index.js

```diff
@@ -0,0 +1,69 @@
+// @ts-nocheck
+
+'use strict';
+
+const _ = require('lodash');
+const matchesStringOrRegExp = require('../../utils/matchesStringOrRegExp');
+const postcss = require('postcss');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const valueParser = require('postcss-value-parser');
+const ruleName = 'function-name-arguments-whitelist';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (property, value) => `Unexpected value "${value}" for property "${property}"`,
+});
+
+function rule(whitelist) {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, {
+			actual: whitelist,
+			possible: [_.isObject],
+		});
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkDecls((decl) => {
+			const prop = decl.prop;
+			const value = decl.value;
+
+			const unprefixedProp = postcss.vendor.unprefixed(prop);
+			const propWhitelist = _.find(whitelist, (list, propIdentifier) =>
+				matchesStringOrRegExp(unprefixedProp, propIdentifier),
+			);
+
+			if (_.isEmpty(propWhitelist)) {
+				return;
+			}
+
+			valueParser(value).walk((node) => {
```

Comment:
checked value parser URL method and got some idea but I don't know to locally test this

Reply:
> but I don't know to locally test this

Paraphasing from the [contributing guide](https://stylelint.io/contributing#run-tests).

You can use the interactive testing prompt to run tests for just a chosen set of files (which you'll want to do during development). For example, to run the tests for just the `function-name-arguments-whitelist` rule:

1. Run `npm run watch` to start the interactive testing prompt.
2. Press `p` to filter by a filename regex pattern.
3. Enter `function-name-arguments-whitelist`
4. Select the file from the list

This will run the tests for the just this rule on every time you save files.

> checked value parser URL method and got some idea

Excellent. Once you have the tests running you can keep trying things out until the tests pass.